File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/VB/WPFListBox-VB/WPFComponents/LanguageDataType.vb
Imports System.Collections.Generic
Imports System.Collections.ObjectModel

Public Class LanguageDataType
  Private mName As String
  Private mLanguageDataTypes As ObservableCollection(Of DataType)

  Public Property Name() As String
    Get
      Return mName
    End Get
    Set(ByVal value As String)
      mName = value
    End Set
  End Property

  Public Property LanguageDataTypes() As ObservableCollection(Of DataType)
    Get
      Return mLanguageDataTypes
    End Get
    Set(ByVal value As ObservableCollection(Of DataType))
      mLanguageDataTypes = value
    End Set
  End Property

  Public Sub New()

  End Sub

  ' Constructor with Param Array
  Public Sub New(ByVal name As String, _
                 ByVal ParamArray dataType As DataType())
    Me.Name = name

    LanguageDataTypes = New ObservableCollection(Of DataType)()
    For Each dt As DataType In dataType
      LanguageDataTypes.Add(dt)
    Next
  End Sub

  Public Function CreateLanguageData() As ObservableCollection(Of LanguageDataType)
    Dim langDataType As New ObservableCollection(Of LanguageDataType)()

    langDataType.Add(New LanguageDataType("C#", New DataType(), New DataType(), New DataType(), New DataType()))

    langDataType.Add(New LanguageDataType("VB.NET", New DataType(), New DataType(), New DataType(), New DataType()))

    Return langDataType
  End Function
End Class

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/LanguageDataType.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace WPFComponents
{
  public class LanguageDataType
  {
    public string Name { get; set; }
    public ObservableCollection<DataType> LanguageDataTypes { get; set; }

    public LanguageDataType()
    {
    }

    // Constructor with Param Array
    public LanguageDataType(string name, params DataType[] dataType)
    {
      this.Name = name;

      LanguageDataTypes = new ObservableCollection<DataType>();
      foreach (DataType dt in dataType)
      {
        LanguageDataTypes.Add(dt);
      }
    }

    public ObservableCollection<LanguageDataType> CreateLanguageData()
    {
      ObservableCollection<LanguageDataType> langDataType = new ObservableCollection<LanguageDataType>();

      langDataType.Add(new LanguageDataType("C#",
          new DataType() { Name = "object" },
          new DataType() { Name = "bool" },
          new DataType() { Name = "string" },
          new DataType() { Name = "int" }));

      langDataType.Add(new LanguageDataType("VB.NET",
          new DataType() { Name = "String" },
          new DataType() { Name = "Integer" },
          new DataType() { Name = "Decimal" },
          new DataType() { Name = "Object" }));

      return langDataType;
    }
  }
}
